### PR TITLE
[SYCL][libclc][E2E] atomic work_item scope fallback

### PR DESCRIPTION
**Pull request by Seanst98**
Add Invocation case, that falls back to a coarser grained scope, to libclc atomic functions. This prevents hangs on AMD and crashes on NVIDIA when using atomic_ref functionality with work_item scope.

Add a test which simply checks that the kernel does not crash when using atomic_ref with work_item scope.

See issue: https://github.com/intel/llvm/issues/16037

### CODE REVIEW

File: sycl/test-e2e/AtomicRef/work_item_scope.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  sycl::device dev;
+  sycl::queue q(dev);
+  auto ctxt = q.get_context();
+
+  // This test does not validate any output
+  // Only that the work_item scope does not error
+  try {
+
+    // Allocate device memory
+    int *data = sycl::malloc_device<int>(1, q);
```
Comment (aelovikov-intel) [inline sycl/test-e2e/AtomicRef/work_item_scope.cpp:20]: This needs to be `free`'ed properly.

Comment (Seanst98) [reply sycl/test-e2e/AtomicRef/work_item_scope.cpp:20]: Good catch. Fixed.

Comment (aelovikov-intel) [reply sycl/test-e2e/AtomicRef/work_item_scope.cpp:20]: > Fixed.

That's why I dislike this part of the SYCL - no RAII for USM :(

Is that common for other e2e tests to "leak" USM on exceptions?

**Review by aelovikov-intel — APPROVED** [review]

**Review by ldrumm — APPROVED** [review]
LGTM

**Pull request merged**